src/libraries/System.Threading.Tasks.Dataflow/src/Base/DataflowBlock.cs

```diff
@@ -2183,7 +2183,7 @@ internal ChooseTarget(StrongBox<Task> completed, CancellationToken cancellationT
                     state =>
                     {
                         var thisChooseTarget = (ChooseTarget<T>)state!;
-                        lock (thisChooseTarget._completed) thisChooseTarget.TrySetCanceled();
+                        lock (thisChooseTarget._completed) thisChooseTarget.TrySetCanceled(cancellationToken);
```

Comment:
This needs to be reverted. Two things:
1. This project compiles for a surface area that doesn't have TrySetCanceled(CancellationToken).
2. Even if that overload were available, this change is problematic and we'd want it to be default.  This is introducing a closure, which means it's adding allocations to code that was explicitly trying to avoid them.  I don't think the analyzer should be flagging this case, where the token comes from outside of the lambda.  Or better yet, it should be a different diagnostic ID so that it can be controlled separately.

Reply:
> I don't think the analyzer should be flagging this case, where the token comes from outside of the lambda. Or better yet, it should be a different diagnostic ID so that it can be controlled separately.

Both options sounds good. I think the case that we do want to ensure is flagged is for invocations inside local functions when token is available in the outer method, as local functions are generally added for readability.

Reply:
> _I think the case that we do want to ensure is flagged is for invocations inside local functions when token is available in the outer method, as local functions are generally added for readability._

Makes sense. So I should remove the ability from the analyzer to suggest passing a token to an invocation inside a lambda, when the lambda itself does not pass a token parameter. Is that correct?

@mavasani if I make this a separate diagnostic ID, can it be added in the same analyzer and fixer classes or should I create new classes?

Reply:
@carlossanlop I think same classes are fine, but you might have to change the tests as probably markup based syntax cannot default to single descriptor? I think fixer should not need any changes as I believe it does not handle this case today.

Reply:
I submitted a proposed fix for the two issues found in this PR. Can you please take a look and let me know if the way I addressed them would be fine? https://github.com/dotnet/roslyn-analyzers/pull/3791

1) If the overload that takes the ct differs by more than 1 parameter, do not offer it in the diagnostic
2) If the invocation that takes a ct is invoked inside a lambda, but the lambda does not pass a ct, do not offer a diagnostic, even if the parent method (local method or normal method) does have a ct parameter.